commands/proxy.ts

```diff
@@ -0,0 +1,102 @@
+import * as commandParams from "../command-params";
+
+export class ProxySetCommand implements ICommand {
+	public disableAnalytics = true;
+
+	public allowedParameters = [new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector)];
+
+	constructor(private $injector: IInjector,
+		private $logger: ILogger,
+		private $prompter: IPrompter,
+		private $proxyCacheService: IProxyCacheService) {
+	}
+
+	public async execute(args: string[]): Promise<void> {
+		let hostname = args[0],
+			port = args[1],
```

Comment:
`--` flags have a semantic meaning of altering a command's behavior which is not the case here. I'm reluctant to use `--` flags in this particular case and I'd rather stick with command-line parameters